lib/persistence.js

```diff
@@ -162,18 +162,31 @@ Persistence.readHashAll = function(hash, callback) {
 };
 
 // Return the value associated with the key *key* (no associated hash)
-Persistence.readKey = function (key, callback) {
+Persistence.readKey = function (key, callback, expireTTL) {
+  var multi = Persistence.redis().multi();
+
   logging.debug('readKey:', key);
-  Persistence.redis().get(key, function (err, reply) {
-    if(err) throw new Error(err);
+  multi.get(key, function (err, reply) {
+    if (err) throw new Error(err);
+    if (expireTTL) {
+      multi.expire(key, expireTTL);
+    }
```

Comment:
A good explanation of the sequence of steps, given the use of multi.  (Curiously, I did a test with the old code, and the key expired as expected.)  Changes made.